### PR TITLE
docs(suspensive.org): add ErrorBoundaryGroup examples using sandpack

### PR DESCRIPTION
**Pull request by kangju2000**
# Overview

Add ErrorBoundaryGroup examples using sandpack.

## PR Checklist

- [x] I did below actions if need

1. I read the [Contributing Guide](https://github.com/toss/suspensive/blob/main/CONTRIBUTING.md)
2. I added documents and tests.

### CODE REVIEW

File: docs/suspensive.org/src/components/Sandpack/baseTemplate.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export default function App() {
     },
   },
   dependencies: {
-    '@tanstack/react-query': '^4.0.0',
```
Comment (kangju2000) [inline docs/suspensive.org/src/components/Sandpack/baseTemplate.ts:27]: @manudeli https://github.com/toss/suspensive/pull/995#discussion_r1657561711

I'm curious if all the features of Suspensive are intended for TanStack Query version 4. 🤔
Are there any features in Suspensive that only work with TanStack Query version 5?

File: docs/suspensive.org/src/components/Sandpack/baseTemplate.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ export default function App() {
...
+    '@tanstack/react-query': '^4',
```
Comment (kangju2000) [inline docs/suspensive.org/src/components/Sandpack/baseTemplate.ts:27]: @manudeli https://github.com/toss/suspensive/pull/995#discussion_r1657561711

I'm curious if all the features of Suspensive are intended for TanStack Query version 4. 🤔
Are there any features in Suspensive that only work with TanStack Query version 5?

Comment (manudeli) [reply docs/suspensive.org/src/components/Sandpack/baseTemplate.ts:27]: > I'm curious if all the features of Suspensive are intended for TanStack Query version 4. 🤔

Yes! but we support @tanstack/react-query@5 too at once by @suspensive/react-query's postinstall script made by @gwansikk. Check these Pull Request out to know how automatic targeting TanStack Query work please if you need.

- https://github.com/toss/suspensive/pull/953
- https://github.com/toss/suspensive/pull/976

When we install, @suspensive/react-query's postinstall script change target of @suspensive/react-query's dist/index.cjs like below

#### if @tanstack/react-query@4' is installed, @suspensive/react-query target @suspensive/react-query-4
```js
dist/index.js
import * from '@suspensive/react-query-4' // default
```

#### if @tanstack/react-query@5' is installed, @suspensive/react-query target @suspensive/react-query-5
```js
dist/index.js
import * from '@suspensive/react-query-5' // changed by postinstall script
```

> Are there any features in Suspensive that only work with TanStack Query version 5?

We support interface like below with following TanStack Query's version

#### @suspensive/react-query with TanStack Query version 4
- useSuspenseQuery
- useSuspenseQueries
- useSuspenseInfiniteQuery
- queryOptions
- SuspenseQuery
- SuspenseQueries
- SuspenseInfiniteQuery
- QueryErrorBoundary

#### @suspensive/react-query with TanStack Query version 5
- SuspenseQuery
- SuspenseQueries
- SuspenseInfiniteQuery
- QueryErrorBoundary


### Conclusion

In my opinion, We should provide Sandpack with TanStack Query v4 yet. but if you disagree it, suggestion also absolutely good for me! Suggest better solution if you have please.

Comment (manudeli) [reply docs/suspensive.org/src/components/Sandpack/baseTemplate.ts:27]: And if you want to get help to know how to use cli of Suspensive React Query.
After you install @suspensive/react-query, you can use `npx suspensive-react-query -h` or `npx srq-h` made by @gwansikk in [this Pull Request with commander.js](https://github.com/toss/suspensive/pull/976)

Comment (kangju2000) [reply docs/suspensive.org/src/components/Sandpack/baseTemplate.ts:27]: > In my opinion, We should provide Sandpack with TanStack Query v4 yet. but if you disagree it, suggestion also absolutely good for me! Suggest better solution if you have please.

The features we offer that are available in @tanstack/react-query@5 can also be used in @tanstack/react-query@4! In that case, it seems we can keep the Sandpack dependency fixed at @tanstack/react-query@4 👍

File: docs/suspensive.org/src/pages/docs/react/ErrorBoundaryGroup.en.mdx [hunk 3]
```diff
@@ -42,22 +103,83 @@ const Example = () => (
...
+            <ErrorBoundaryGroup.Consumer>
+              {(group) => <button onClick={group.reset}>Try again</button>}
+            </ErrorBoundaryGroup.Consumer>
+            <ErrorBoundary fallback={(props) => <>{props.error.message}</>}>
+              <ErrorAfter2s />
+            </ErrorBoundary>
+          </fieldset>
+        </ErrorBoundaryGroup>
+      </fieldset>
+    </ErrorBoundaryGroup>
```
Comment (manudeli) [inline docs/suspensive.org/src/pages/docs/react/ErrorBoundaryGroup.en.mdx:153]: So much better experience....!!! 🚀🚀🚀🚀🚀

image

**Review by manudeli — APPROVED** [review]

File: docs/suspensive.org/src/components/Sandpack/baseTemplate.ts (hunk 2 above)
Comment (manudeli) [reply docs/suspensive.org/src/components/Sandpack/baseTemplate.ts:27]: > In that case, it seems we can keep the Sandpack dependency fixed at @tanstack/react-query@4 👍

That's right~!

**Pull request merged**